be/src/vec/exec/vunion_node.cpp

```diff
@@ -220,6 +221,21 @@ Status VUnionNode::get_next_const(RuntimeState* state, Block* block) {
     return Status::OK();
 }
 
+//for pipeline operator
+Status VUnionNode::materialize_child_block(RuntimeState* state, int child_id,
+                                           vectorized::Block* input_block,
+                                           vectorized::Block* output_block) {
+    DCHECK_LT(child_id, _children.size());
+    DCHECK(!is_child_passthrough(child_id));
+    MutableBlock mblock = MutableBlock(
```

Comment:
rethink the mem reuse and try to merge the block to make sure output_block have `batch_size`.